napari/_qt/layer_controls/qt_shapes_controls.py

```diff
@@ -91,7 +91,7 @@ def __init__(self, layer):
         self.layer.events.current_face_color.connect(
             self._on_current_face_color_change
         )
-        self.layer.events.editable.connect(self._on_editable_change)
+        self.layer.events.editable.connect(self._on_editable_or_visible_change)
```

Comment:
As above - is this missing?
```suggestion
        self.layer.events.editable.connect(self._on_editable_or_visible_change)
        self.layer.events.visible.connect(self._on_editable_or_visible_change)
```